File: src/main/kotlin/com/ederfmatos/reactive/java/SyncOneSerializer.java
package com.ederfmatos.reactive.java;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;

public class SyncOneSerializer extends StdSerializer<SyncOne<?>>{

    public SyncOneSerializer() {
        this(null);
    }

    protected SyncOneSerializer(Class<SyncOne<?>> t) {
        super(t);
    }

    @Override
    public void serialize(SyncOne<?> syncOne, JsonGenerator jsonGenerator, SerializerProvider provider) throws IOException {
        jsonGenerator.writeObject(syncOne.value);
    }
}

File: src/main/kotlin/com/ederfmatos/reactive/kotlin/SyncOneSerializer.kt
package com.ederfmatos.reactive.kotlin

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdSerializer

class SyncOneSerializer : StdSerializer<SyncOne<*>>(SyncOne::class.java) {
    override fun serialize(syncOne: SyncOne<*>, jsonGenerator: JsonGenerator, provider: SerializerProvider?) {
        jsonGenerator.writeObject(syncOne.value)
    }
}